eslint.config.js

```diff
@@ -34,6 +34,32 @@ export default tsEslint.config(
       "@typescript-eslint/consistent-type-exports": "error",
       "@typescript-eslint/no-import-type-side-effects": "error",
       "@typescript-eslint/consistent-type-imports": "error",
+      "@typescript-eslint/use-unknown-in-catch-callback-variable": "off",
+      "@typescript-eslint/restrict-template-expressions": [
+        "error",
+        {
+          allowAny: true,
+          allowBoolean: true,
+          allowNullish: false,
+          allowNumber: true,
+          allowRegExp: true,
+        },
+      ],
+      "@typescript-eslint/no-confusing-void-expression": [
+        "error",
+        {
+          ignoreArrowShorthand: true,
+          ignoreVoidOperator: true,
+        },
+      ],
```

Comment:
Ok, I started my journey here without the overrides. And then like 50 files in... it started to feel a bit overboard... 

The core of this rule supports lots of good patterns, but for these 2 overrides, i feel like I'm comfortable not going all the way.

This means, we can still:
```ts
<button onClick={() => console.log("🎉")} />
```

https://typescript-eslint.io/rules/no-confusing-void-expression/